### ISSUE TITLE
Plugin examples

### ISSUE THREAD

**Issue by Crazee** [Expensify/Bedrock#224]
What would really help are a couple of examples on how to write (and compile and load) a plugin that defines a sqllite stored procedure or function.

If you currently use a couple of mySQL "CREATE FUNCTION ..." procedures a "simple database conversion" won't get you there.

**Comment by quinthar** [comment]
Fair point.  I'd suggest the simplest example is the [Bedrock::DB plugin](https://github.com/Expensify/Bedrock/blob/master/plugins/DB.cpp), which is really just a single .h/cpp file pair.  Don't get much more simple than that!

As for compiling it, it does require creating a shared library, and I'm not sure if we have any good examples here.  @tylerkaraszewski can you provide some guidance here?

Thanks, and sorry for the slow response!
